cmd/gardener-metrics-exporter/app.go

```diff
@@ -122,30 +124,45 @@ func run(o *options, closeCh chan os.Signal) error {
 	return nil
 }
 
+func newConfigFromBytes(kubeconfig string) (*restclient.Config, error) {
+	kubecf, err := ioutil.ReadFile(kubeconfig)
+	if err != nil {
+		log.Errorf("Could not read config")
+	}
+	configObj, err := clientcmd.Load(kubecf)
+	if err != nil {
+		log.Errorf("Could not load config")
+	}
+	clientConfig := clientcmd.NewDefaultClientConfig(*configObj, &clientcmd.ConfigOverrides{})
+	client, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, nil
+	}
+	if client == nil {
+		return nil, err
+	}
+	return client, nil
+}
+
 func setupInformerFactories(kubeconfigPath string, stopCh <-chan struct{}) (gardeninformers.SharedInformerFactory, kubeinformers.SharedInformerFactory, error) {
-	k8sGardenClient, err := kubernetes.NewClientFromFile(kubeconfigPath)
+	restConfig, err := newConfigFromBytes(kubeconfigPath)
 	if err != nil {
 		return nil, nil, err
 	}
-
-	gardenerClient := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		&clientcmd.ClientConfigLoadingRules{ExplicitPath: kubeconfigPath},
-		&clientcmd.ConfigOverrides{},
-	)
-
-	gardenerClientConfig, err := gardenerClient.ClientConfig()
+	if restConfig == nil {
+		return nil, nil, err
+	}
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
```

Comment:
This was already before an potential issue, but `k8sClient` and `gardenClient` can be nil. Could you please add some checks to avoid that?

Reply:
Checks are added.